src/librustc/ty/mod.rs

```diff
@@ -1107,7 +1107,15 @@ impl<'tcx> Predicate<'tcx> {
                 vec![]
             }
             ty::Predicate::Projection(ref data) => {
-                let trait_inputs = data.0.projection_ty.trait_ref.input_types();
+                // FIXME(tschottdorf): lost the fight against the borrow checker here,
+                // at least for now.
+                //
+                // let trait_inputs =
+                //   ty::tls::with(|tcx| data.0.projection_ty.trait_ref(tcx)).input_types();
+                //   --------------------------------------------------------              ^
+                //   |                     temporary value dropped here while still borrowed
+                //   temporary value created here
+                let trait_inputs = data.0.projection_ty.deprecated_trait_ref.input_types();
                 trait_inputs.chain(Some(data.0.ty)).collect()
```

Comment:
Done (assume you mean changing to `data.0.projection_ty.substs.types().chain(Some(data.0.ty)).collect()`